### ISSUE TITLE
Multiple Joins in Spark3 Dialect

### ISSUE THREAD

**Issue by RaymondMcT** [sqlfluff/sqlfluff#2916]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The spark3 dialect fails to parse multiple joins. The same simple sql is parsable with ansi, postgres and bigquery dialects (the ones I checked).

### Expected Behaviour

It should be parsable.

### Observed Behaviour

If you perform multiple JOINS the parser fails. These can be of varying types. 

### How to reproduce

```sql
-- test.sql
SELECT
    table1.a,
    table2.b,
    table3.c
FROM table1
INNER JOIN table2
    ON table1.a = table2.a
INNER JOIN table3
    ON table1.a = table3.a
```

```bash
sqlfluff lint --dialect spark3 test.sql
```

```
== [test.sql] FAIL                                                                                                                                                                   
L:   7 | P:  27 |  PRS | Line 7, Position 27: Found unparsable section: '\nINNER
                       | JOIN table3\n ON table1.a = tab...'
```

### Dialect

spark3

### Version

sqlfluff, version 0.11.1
python3 3.8.13

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
Fixed in #2917 . Planning a release tomorrow and this should make it into that (FYI @alanmcruickshank )

**Comment by RaymondMcT** [comment]
9 minutes. Hero. 👍

**Comment by tunetheweb** [comment]
I’m procrastinating work I should be doing 😁
